app/component.app.ts

```diff
@@ -10,15 +10,17 @@ export class Artist {
 
 @Component({
   selector: 'app',
-  templateUrl: './partials/app.html',
-  styleUrls: [ "./css/app.css" ] 
+  templateUrl: '../partials/app.html',
+  styleUrls: [ "../css/app.css" ] 
 })
 
 export class AppComponent {
   artists = ARTISTS;
   currentArtist: Artist;
 
-  showArtist(item) {
+  @Input() query: string = null;
```

Comment:
I was getting some errors related to undefined inputs on your components. I don't know much about the `@Input` annotation but I came across it in a StackOverflow post (lost the link...), and it guided me to [this example](https://plnkr.co/edit/VSctCwNMctHIp4WuHLBy?p=preview) which fixed the problem. It looks like you need to define the type of each input like this.